### PR TITLE
converting to single expression

### PR DESCRIPTION
**Pull request by ayushhooda**

### CODE REVIEW

**Comment by ayushhooda** [comment]
@rossabaker Thanks
I have corrected the scalafmt check. Please see if it's good to go now

**Comment by ayushhooda** [comment]
@rossabaker Could you please add the second reviewer.

File: circe/src/main/scala/org/http4s/circe/middleware/JsonDebugErrorHandler.scala [hunk 1]
```diff
@@ -75,13 +75,11 @@ object JsonDebugErrorHandler {
...
         redactWhen: CIString => Boolean
     ): Encoder[JsonErrorHandlerResponse[F]] =
-      new Encoder[JsonErrorHandlerResponse[F]] {
-        def apply(a: JsonErrorHandlerResponse[F]): Json =
-          Json.obj(
-            "request" -> encodeRequest(a.req, redactWhen),
-            "throwable" -> encodeThrowable(a.caught)
-          )
-      }
+      (a: JsonErrorHandlerResponse[F]) =>
```
Comment (kevinmeredith) [inline circe/src/main/scala/org/http4s/circe/middleware/JsonDebugErrorHandler.scala:78]: Could you please say your rationale for making this change, @ayushhooda? Do you find it easier to read, more performant, etc.? 

circe's [Custom encoders/decoders](https://circe.github.io/circe/codecs/custom-codecs.html#custom-encodersdecoders) points to the existing approach, but I admit that I don't know why that is, i.e. as opposed to the approach you've taken in this PR.

Thanks

Comment (ayushhooda) [reply circe/src/main/scala/org/http4s/circe/middleware/JsonDebugErrorHandler.scala:78]: @kevinmeredith
In the majority of cases, lambda expressions could easily replace anonymous classes.
It makes code more **functional**, **precise**, and **readable**.

Comment (kevinmeredith) [reply circe/src/main/scala/org/http4s/circe/middleware/JsonDebugErrorHandler.scala:78]: Thanks for providing your rationale!

**Comment by ayushhooda** [comment]
@rossabaker Please add the second reviewer as I'm not very sure who would be the best person to review this.

**Comment by ayushhooda** [comment]
Thanks, @hamnis for approval
@rossabaker Are we good to merge this?

**Pull request merged**